strawberry/enum.py

```diff
@@ -11,45 +12,69 @@ class EnumValue:
     value: Any
 
 
-@dataclasses.dataclass
-class EnumDefinition:
-    name: str
-    values: List[EnumValue]
-    description: Optional[str]
+class StrawberryEnum:
+    def __init__(
+        self,
+        enum: EnumMeta,
+        python_name: str,
+        graphql_name: Optional[str],
+        description: Optional[str],
+    ) -> None:
+        self.enum = enum
+        self.python_name = python_name
+        self._graphql_name = graphql_name
+        self.description = description
+
+    def __call__(self, *args: Any, **kwds: Any) -> Any:
+        return self.enum(*args, **kwds)
+
+    def __getattr__(self, attr):
+        if hasattr(self.enum, attr):
+            return self.enum[attr]
+
+        return super().__getattribute__(attr)
+
+    @property
+    def values(self) -> List[EnumValue]:
+        return [
+            EnumValue(item.name, item.value) for item in cast(Iterable[Enum], self.enum)
+        ]
+
+    @property
+    def graphql_name(self) -> str:
+        if self._graphql_name:
+            return self._graphql_name
+
+        return to_camel_case(self.python_name)
 
 
 def _process_enum(
     cls: EnumMeta, name: Optional[str] = None, description: Optional[str] = None
-) -> EnumMeta:
+) -> StrawberryEnum:
     if not isinstance(cls, EnumMeta):
         raise NotAnEnum()
 
     if not name:
         name = cls.__name__
 
-    description = description
-
-    values = [EnumValue(item.name, item.value) for item in cls]  # type: ignore
-
-    cls._enum_definition = EnumDefinition(  # type: ignore
-        name=name,
-        values=values,
+    return StrawberryEnum(
+        enum=cls,
+        python_name=cls.__name__,
+        graphql_name=name,
         description=description,
     )
 
-    return cls
-
 
 def enum(
     _cls: EnumMeta = None, *, name=None, description=None
-) -> Union[EnumMeta, Callable[[EnumMeta], EnumMeta]]:
+) -> Union[StrawberryEnum, Callable[[EnumMeta], StrawberryEnum]]:
```

Comment:
Why does this take a `Callable`?

Reply:
Why would you pass a `StrawberryEnum` to this function?

Reply:
this is the return type :)